source/linux/Foundational_Components/U-Boot/UG-Memory.rst

```diff
@@ -1,973 +1,1120 @@
+   Booting Linux from USB storage
+   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    .. code-block:: console
+   To load the Linux Kernel and rootfs from USB rather than SD/MMC card on
+   AMx/DRA7x EVMs, if we assume that the USB device is partitioned the same
+   way as an SD/MMC card is, we can utilize the :command:`mmcboot` command
+   to boot. To do this, perform the following steps:
 
-        U-Boot # usb start
-        U-Boot # setenv mmcroot /dev/sda2 ro
-        U-Boot # run mmcargs
-        U-Boot # run bootcmd_usb
+   .. code-block:: console
 
-    On K2H/K/E/L EVMs, the USB drivers in Kernel needs to be built-in
-    (default modules). The configuration changes are:
+      => usb start
+      => setenv mmcroot /dev/sda2 ro
+      => run mmcargs
+      => run bootcmd_usb
 
-    .. code-block:: text
+   On K2H/K/E/L EVMs, the USB drivers in Linux kernel need to be built-in
+   (default modules). The configuration changes are:
 
-        CONFIG_USB=y
-        CONFIG_USB_XHCI_HCD=y
-        CONFIG_USB_XHCI_PCI=y
-        CONFIG_USB_XHCI_PLATFORM=y
-        CONFIG_USB_STORAGE=y
-        CONFIG_USB_DWC3=y
-        CONFIG_USB_DWC3_HOST=y
-        CONFIG_USB_DWC3_KEYSTONE=y
-        CONFIG_EXTCON=y
-        CONFIG_EXTCON_USB_GPIO=y
-        CONFIG_SCSI_MOD=y
-        CONFIG_SCSI=y
-        CONFIG_BLK_DEV_SD=y
+   .. code-block:: Kconfig
 
-    The USB should have boot partition of FAT32 format, and rootfs partition
-    of EXT4 format. The boot partition must contain the following images:
+      CONFIG_USB=y
+      CONFIG_USB_XHCI_HCD=y
+      CONFIG_USB_XHCI_PCI=y
+      CONFIG_USB_XHCI_PLATFORM=y
+      CONFIG_USB_STORAGE=y
+      CONFIG_USB_DWC3=y
+      CONFIG_USB_DWC3_HOST=y
+      CONFIG_USB_DWC3_KEYSTONE=y
+      CONFIG_EXTCON=y
+      CONFIG_EXTCON_USB_GPIO=y
+      CONFIG_SCSI_MOD=y
+      CONFIG_SCSI=y
+      CONFIG_BLK_DEV_SD=y
 
-    .. code-block:: text
+   The USB should have boot partition of FAT32 format, and rootfs partition
+   of EXT4 format. The boot partition must contain the following images:
 
-        keystone-<platform>-evm.dtb
-        skern-<platform>.bin
-        k2-fw-initrd.cpio.gz
-        zImage
+   - keystone-<platform>-evm.dtb
+   - skern-<platform>.bin
+   - k2-fw-initrd.cpio.gz
+   - zImage
 
-        where <platform>=k2hk, k2e, k2l
+   where <platform>=k2hk, k2e, k2l
 
-    The rootfs partition contains the filesystem from ProcSDK release
-    package.
+   The rootfs partition contains the filesystem from ProcSDK release package.
 
-    .. code-block:: console
+   .. code-block:: console
 
-        # mkdir /mnt/temp
-        # mount -t ext4 /dev/sdb2 /mnt/temp
-        # cd /mnt/temp
-        # tar xvf <Linux_Proc_Sdk_Install_DIR>/filesyste/tisdk-server-rootfs-image-k2hk-evm.tar.xz
-        # cd /mnt
-        # umount temp
+      $ mkdir /mnt/temp
+      $ mount -t ext4 /dev/sdb2 /mnt/temp
+      $ cd /mnt/temp
+      $ tar xvf <Linux_Proc_Sdk_Install_DIR>/filesyste/tisdk-server-rootfs-image-k2hk-evm.tar.xz
```

Comment:
Should use sudo or other method to ensure original file permissions and user/group are preserved.

Reply:
Done